spell/rwa-checklists.md

```diff
@@ -0,0 +1,191 @@
+        * [`DssExecLib.increaseGlobalDebtCeiling(amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L436)
+        * [`DssExecLib.decreaseGlobalDebtCeiling(amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L445C14-L445C39)
+  * [ ] Liquidation oracle price is bumped via `RwaLiquidationOracleLike(MIP21_LIQUIDATION_ORACLE).bump(ilk, val)` pattern
+    * [ ] Comment above `bump` explains `val` computation via `// Note: the formula is: "debt_ceiling * [ (1 + rwa_stability_fee ) ^ (minimum_deal_duration_in_years) ] * liquidation_ratio"`
+    * [ ] Comment  above `bump` provides locally executable formula (e.g. `// bc -l <<< 'scale=18; 50000000 * e(l(1.07) * (3342/365)) * 1.00' | cast --to-wei`)
+      * [ ] The formula matches the example provided above
+      * [ ] `debt_ceiling` in the executable formula matches new debt ceiling set in the spell or the maximum possible debt ceiling in case of the enabled AutoLine
+      * [ ] `rwa_stability_fee` in the executable formula matches stability fee of the specified RWA found on chain
+      * [ ] `minimum_deal_duration_in_years` in the executable formula matches number found in the Exec Sheet of the spell containing relevant RWA onboarding
+      * [ ] `liquidation_ratio` in the executable formula matches liquidation ratio of the specified RWA found on chain
+      * [ ] Executing formula locally provides integer number that matches the `val` in the spell
+    * [ ] `val` makes sense in context of the [rate mechanism](https://github.com/makerdao/developerguides/blob/master/mcd/intro-rate-mechanism/intro-rate-mechanism.md)
+  * [ ] IF multiple RWA ilks are being combined into one, `val` calculation is done once per ilk and added to make the total, with separate executable formulas provided in comments. The existing `val` value can be retrieved by calling `read()` on `PIP_RWAXYZ` and converting the result into decimal
+  * [ ] Oracle price is updated via `DssExecLib.updateCollateralPrice(ilk)`
+  * IF soft liquidation explicitly requested to be triggered (via `.tell(ilk)`) AND debt ceiling (`vat.ilks(ilk).line`) is `0` (OR is being set to `0` in the current spell)
+      * [ ] `RwaLiquidationOracle.tell(ilk)` call is present
+      * [ ] IF `RWAXYZ_A_INPUT_CONDUIT` is an instance of [`TinlakeMgr`](https://github.com/centrifuge/tinlake-maker-lib/blob/master/src/mgr.sol) (it is a Centrifuge integration), additional `TinlakeMgr.tell()` call is present (in order to prevent further `TIN` redemptions in the Centrifuge pool)
+
+## RWA Offboarding Checklist
+
+* IF there is debt in the RWA Vault (`vat.urns(ilk, RWAXYZ_A_URN).art > 0`), proceed with the write-off (i.e.: vault is in default):
+  * IF soft liquidation `.tell(ilk)` has **NOT** been called for the ilk in a previous spell:
+    * [ ] The value of `RwaLiquidationOracle.ilks(ilk).tau` MUST be zero, OTHERWISE the offboarding can only happen in a future spell.
```

Comment:
Can the rest of checklist still happen except `RwaLiquidationOracle.cull(ilk, RWAXYZ_A_URN)`? (eg. `Set Debt ceiling to 0`) If then I'd prefer further specify to avoid confusion.

Reply:
Yeah, it makes sense. I'll amend this.

Reply:
Please check now.